### ISSUE TITLE
Correct and Smooth implementation uses validation labels 

### ISSUE THREAD

**Issue by LucasPrietoAl** [dmlc/dgl#4102]
## 🐛 Bug

The dgl implementation of Correct & Smooth (https://github.com/dmlc/dgl/tree/master/examples/pytorch/correct_and_smooth) uses validation labels for post processing. The accuracy is compared to that of the author's code which does not use the validation labels by default. In general, the version of Correct & Smooth which is accepted for benchmarking datasets like OGB is the one that does not use validation labels.

## Expected behavior

Default Correct and Smooth predictions should not use validation labels. 

## Environment

N/A

## Additional context

This is the problematic line in main.py:
mask_idx = torch.cat([train_idx, valid_idx])

mask_idx should be replaced by train_idx in main.py or at least that should be the default

**Comment by mufeili** [comment]
Nice catch! Have you tried re-performing the experiments without using the validation labels? Do you observe a significant performance drop?

**Comment by LucasPrietoAl** [comment]
Yes, the validation labels make a big difference. Without them the test accuracy goes down from 72.6% to 68.67% for Linear+ C&S on arxiv. Using the same alpha values as the authors this goes back up to 70%, but this is still below the accuracy of the equivalent model from the authors (71.26%). When I have a bit more time I will look at what could be the reason behind this difference in accuracy.

Also, I assume leaving out the spectral embeddings from this method was intentional, but this results in a disadvantageous comparison for the Author's model as their hyperparameters seem to be tuned to use spectral embeddings.

**Comment by LucasPrietoAl** [comment]
Ok, I should be able to help with that in a couple of days
